woocommerce/woocommerce
# Marketplace: table design updates

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:
Brings design updates to the my subscriptions table. 
- Update table headers and descriptions. The refresh button is now standalone and not a link in the description. 
- Removed the third free-to-install table
- Merges status and product name columns and adds a badge for not connected products
- Add a new element for viewing documentation 

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Build WooCommerce Admin assets by `cd plugins/woocommerce-admin && pnpm run start`
2. Visit the My Subscription page and see the design changes. 
3. Clicking the new **Refresh** button should still work.
4. It isn't likely you won't have expired subscriptions. To make them show up, you may add this snippet into the `WC_Helper::get_subscription_list_data`. This also starts showing the **View Documentation** action dropdown item. You should place some product keys into the arrays defined below. I suggest picking extensions that are installed and not installed. So you see actions in both tables. 

```php
// Make expired. Shows renew button.
$subscriptions_to_expire = [
	'INSERT_PRODUCT_KEY_HERE', // Product keys start with W00. 
];

// Remove product key. Shows subscribe button.
$subscriptions_to_remove_licence = [
	'INSERT_PRODUCT_KEY_HERE', // Product keys start with W00. 
];

foreach ($subscriptions as &$subscription) {
	if ( in_array( $subscription['product_key'], $subscriptions_to_expire, true ) ) {
		$subscription['expired'] = true;
		$subscription['expires'] = 100;
		$subscription['documentation_url'] = 'https://docs.woocommerce.com/document/example/';
	}

	if ( in_array( $subscription['product_key'], $subscriptions_to_remove_licence, true ) ) {
		$subscription['product_key'] = '';
                $subscription['documentation_url'] = 'https://docs.woocommerce.com/document/example/';
	}
}
```

### Screenshot
![image](https://github.com/woocommerce/woocommerce/assets/10389957/95fc2fbb-735b-4135-9576-ec4bb3d60f00)
